Prototope/ShapeLayer.swift

```diff
@@ -0,0 +1,424 @@
+	public var fillColor: Color? = Color.black {
+		didSet {
+			shapeViewLayerStyleChanged()
+		}
+	}
+	
+	
+	/** The stroke colour for the shape. Defaults to `Color.black`. */
+	public var strokeColor: Color? = Color.black {
+		didSet {
+			shapeViewLayerStyleChanged()
+		}
+	}
+	
+	
+	/** The width of the stroke. Defaults to 1.0. */
+	public var strokeWidth = 1.0 {
+		didSet {
+			shapeViewLayerStyleChanged()
+		}
+	}
+	
+	
+	/** If the path is closed, the first and last segments will be connected. */
+	public var closed: Bool {
+		didSet {
+			self.setNeedsDisplay()
+		}
+	}
+	
+	
+	/** Represents the types of cap styles path segment endpoints will show. Only affects open paths. */
+	public enum LineCapStyle {
+		
+		/** The line cap will have butts for ends. */
+		case Butt
+		
+		/** The line cap will have round ends. */
+		case Round
+		
+		/** The line cap will have square ends. */
+		case Square
+		
+		internal func capStyleString() -> String {
+			switch self {
+			case Butt:
+				return kCALineCapButt
+			case Round:
+				return kCALineCapRound
+			case Square:
+				return kCALineCapSquare
+			}
+		}
+	}
+	
+	
+	/** The line cap style for the path. Defaults to LineCapStyle.Butt. */
+	public var lineCapStyle: LineCapStyle = .Butt {
+		didSet {
+			shapeViewLayerStyleChanged()
+		}
+	}
+	
+	
+	/** Represents the types of join styles path segments will show at their joins. */
+	public enum LineJoinStyle {
+		
+		/** Lines will be joined with a miter style. */
+		case Miter
+		
+		/** Lines will be joined with a round style. */
+		case Round
+		
+		/** Line joins will have beveled edges. */
+		case Bevel
+		
+		internal func joinStyleString() -> String {
+			switch self {
+			case Miter:
+				return kCALineJoinMiter
+			case Round:
+				return kCALineJoinRound
+			case Bevel:
+				return kCALineJoinBevel
+			}
+		}
+	}
+	
+	
+	/** The line join style for path lines. Defaults to LineJoinStyle.Miter. */
+	public var lineJoinStyle: LineJoinStyle = .Miter {
+		didSet {
+			shapeViewLayerStyleChanged()
+		}
+	}
+	
+	
+	// MARK: - Private details
+	
+	private var shapeViewLayer: CAShapeLayer {
+		return self.view.layer as! CAShapeLayer
+	}
+	
+	
+	private class ShapeView: UIView {
+		override class func layerClass() -> AnyClass {
+			return CAShapeLayer.self
+		}
+	}
+	
+	
+	private func shapeViewLayerStyleChanged() {
+		var layer = self.shapeViewLayer
+		layer.lineCap = self.lineCapStyle.capStyleString()
+		layer.lineJoin = self.lineJoinStyle.joinStyleString()
+		
+		if let fillColor = fillColor {
+			self.shapeViewLayer.fillColor = fillColor.CGColor
+		} else {
+			self.shapeViewLayer.fillColor = nil
+		}
+		
+		
+		if let strokeColor = strokeColor {
+			self.shapeViewLayer.strokeColor = strokeColor.CGColor
+		} else {
+			self.shapeViewLayer.strokeColor = nil
+		}
+		
+		self.shapeViewLayer.lineWidth = CGFloat(strokeWidth)
+	}
+	
+	
+	private var bezierPath: UIBezierPath {
```

Comment:
This is the kind of code that could really use some unit tests. Lots of edge cases here—what if there's only one segment on a closed path? What if there are no segments? What if we alternate straight and curved segments? Etc